xmake/core/project/target.lua

```diff
@@ -416,9 +416,10 @@ function _instance:get_from_pkgs(name, opt)
         local configinfo = self:pkgconfig(pkg:name())
         -- get values from package components
         -- e.g. `add_packages("sfml", {components = {"graphics", "window"}})`
-        if configinfo and configinfo.components and pkg:components() then
+        local components_default = pkg:components_default()
+        if configinfo and (configinfo.components or components_default) and pkg:components() then
```

Comment:
this won't work if configinfo is nil (which is the case with `add_packages("xx")`

this works:
```lua
        local selected_components = configinfo and configinfo.components or pkg:components_default()
        if selected_components and pkg:components() then
            local components_enabled = hashset.new()
            for _, comp in ipairs(table.wrap(selected_components)) do
```

Reply:
try it again.

Reply:
works fine, thanks! 👍